lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -0,0 +1,408 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_server
+
+short_description: Create and manage cloud servers on the Hetzner Cloud
+
+version_added: "2.8"
+
+description:
+    - "Create, update and manage cloud servers on the Hetzner Cloud."
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
+
+options:
+    id:
+        description:
+            - The Server ID
+            - Only required if no server name is given
+        type: int
+    name:
+        description:
+            - The Server Name
+            - Only required if no server id is given or a server does not exists
+        type: str
+    server_type:
+        description:
+            - The Server Type
+            - Required if server does not exists
+        type: str
+    ssh_keys:
+        description:
+            - List of SSH Keys Names
+        type: list
+        aliases: [ ssh_key ]
```

Comment:
I would not add this alias. We tend to use aliases for backward compatibility or to add consistency between modules. But there's no value in adding this alias here. Rather stick to one parameter name.

Reply:
This was a suggestion from @resmo.

I think it is more intuitive for some people to write
```yaml
ssh_key: my-ssh-key
```
instead of 
```yaml
ssh_keys: my-ssh-key
```
or 
```yaml
    ssh_keys:
      - my-ssh-key
```

My suggestion would be to leave it, because some other cloud modules have the same behavior (vultr as sample)

Reply:
All our parameters that accept lists, accept single items too. But we never offer both the plural and singular version for parameter name. A one-item list, is still a list :-)

So I wouldn't bother. IMO offering both is more confusing as people may assume that:
```yaml
    ssh_key: my-ssh-key
```
means there is only one SSH key to add. That is why I would only have the plural form and no aliases. It will end up to be more consistent between users/playbooks and it's clear the parameter accepts more than one value.

Reply:
I clarified my position above, and this will spark a discussion. Don't worry, the best way to settle these cases once and for all is by doing them on the spot ;-) Let's hope we come to a consensus soon.

Reply:
As sample where it is with an singular alias too:
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vultr/vultr_server.py#L82

Reply:
We have many modules doing things we do not recommend or wouldn't recommend today. In any case there are more examples where we don't have singular parameter aliases for plural parameters (lists) than where we do it.

The most important reason for me is that it adds to the confusion, people expecting a single SSH key value is possible because the parameter used is singular. (Yes, they should have read the documentation, but lots of users don't and the singular form *is* deceiving)